articles/dms/howto-mysql-to-azure-mysql-powershell.md

```diff
@@ -0,0 +1,657 @@
+    $optionalAgentSettings | add-member -MemberType NoteProperty -Name $settingName -Value $settingValue
+}
+
+# Set any optional settings in the input based on parameters to this cmdlet
+AddOptionalSetting $offlineMigTaskProperties.input.optionalAgentSettings "DesiredRangesCount" $DesiredRangesCount;
+AddOptionalSetting $offlineMigTaskProperties.input.optionalAgentSettings "MaxBatchSizeKb" $MaxBatchSizeKb;
+AddOptionalSetting $offlineMigTaskProperties.input.optionalAgentSettings "MinBatchRows" $MinBatchRows;
+AddOptionalSetting $offlineMigTaskProperties.input.optionalAgentSettings "PrepareDatabaseForBulkImportTaskCount" $PrepareDatabaseForBulkImportTaskCount;
+AddOptionalSetting $offlineMigTaskProperties.input.optionalAgentSettings "PrepareTableForBulkImportTaskCount" $PrepareTableForBulkImportTaskCount;
+AddOptionalSetting $offlineMigTaskProperties.input.optionalAgentSettings "QueryTableDataRangeTaskCount" $QueryTableDataRangeTaskCount;
+AddOptionalSetting $offlineMigTaskProperties.input.optionalAgentSettings "WriteDataRangeBatchTaskCount" $WriteDataRangeBatchTaskCount;
+AddOptionalSetting $offlineMigTaskProperties.input.optionalAgentSettings "MaxBatchCacheSizeMb" $MaxBatchCacheSizeMb;
+AddOptionalSetting $offlineMigTaskProperties.input.optionalAgentSettings "ThrottleQueryTableDataRangeTaskAtAvailableMemoryMb" $ThrottleQueryTableDataRangeTaskAtAvailableMemoryMb;
+AddOptionalSetting $offlineMigTaskProperties.input.optionalAgentSettings "ThrottleQueryTableDataRangeTaskAtBatchCount" $ThrottleQueryTableDataRangeTaskAtBatchCount;
+AddOptionalSetting $offlineMigTaskProperties.input.optionalAgentSettings "DelayProgressUpdatesInStorageInterval" $DelayProgressUpdatesInStorageInterval;
+```
+
+## Step 11: Running the migration task
+
+After configuration of the task, you will now execute the task and wait for the migration to complete.
+
+The following script invokes the configured migration task and waits for it to complete.
+
+```powershell
+# Running the migration scenario
+[string] $TaskName = "mysqlofflinemigrate"
+
+LogMessage -Message "Running data migration scenario ..." -IsProcessing $true
+$summary = @{
+    "SourceServer"   = $SourceServerName;
+    "SourceDatabase" = $SourceDatabaseName;
+    "TargetServer"   = $TargetServerName;
+    "TargetDatabase" = $TargetDatabaseName;
+    "TableCount"     = $tableMap.Count;
+    "StartedOn"      = $offlineMigTaskProperties.input.startedOn;
+}
+
+Write-Host "Job Summary:" -ForegroundColor Yellow
+Write-Host $(ConvertTo-Json $summary) -ForegroundColor Yellow
+
+$migrationResult = RunScenario -MigrationService $dmsService `
+    -MigrationProject $dmsProject `
+    -ScenarioTaskName $TaskName `
+    -TaskProperties $offlineMigTaskProperties
+
+LogMessage -Message "Migration completed with status - $($migrationRsult.state)"
+#Checking for any errors or warnings captured by the task during migration
+$dbLevelResult = $migrationResult.output | Where-Object { $_.resultType -eq "DatabaseLevelOutput" } 
```

Comment:
You also have to check for task level errors, which can be done via $migrationResult.errors.details

Reply:
Done